XNAControls/XNAHorizontalScrollBar.cs

```diff
@@ -0,0 +1,343 @@
+        thumbLeft = AssetLoader.LoadTexture("hsbThumbLeft.png");
+        thumbRight = AssetLoader.LoadTexture("hsbThumbRight.png");
+    }
+
+    /// <summary>
+    /// Scrolls up when the user presses on the "scroll left" arrow.
+    /// </summary>
+    private void BtnScrollLeftLeftClick(object sender, EventArgs e)
+    {
+        if (ViewLeft > 0)
+        {
+            ViewLeft -= ScrollStep;
+            if (ViewLeft < 0)
+                ViewLeft = 0;
+        }
+
+        RefreshButtonX();
+
+        Scrolled?.Invoke(this, EventArgs.Empty);
+    }
+
+    /// <summary>
+    /// Scrolls down when the user presses on the "scroll down" arrow.
+    /// </summary>
+    private void BtnScrollRightLeftClick(object sender, EventArgs e)
+    {
+        int nonDisplayedLines = Length - DisplayedPixelCount;
+
+        if (ViewLeft < nonDisplayedLines)
+            ViewLeft = Math.Min(ViewLeft + ScrollStep, nonDisplayedLines);
+
+        RefreshButtonX();
+
+        Scrolled?.Invoke(this, EventArgs.Empty);
+    }
+
+    /// <summary>
+    /// Returns a bool that tells whether there's enough items in a list for 
+    /// the scrollbar to be drawn.
+    /// </summary>
+    public bool IsDrawn()
+    {
+        return scrollablePixels > 0;
+    }
+
+    /// <summary>
+    /// Refreshes the scroll bar's thumb size.
+    /// </summary>
+    public void Refresh()
+    {
+        int width = Width -
+            btnScrollLeft.Width - btnScrollRight.Width;
+
+        int nonDisplayedColumns = Length - DisplayedPixelCount;
+
+        if (nonDisplayedColumns <= 0)
+        {
+            thumbWidth = width;
+            scrollablePixels = 0;
+            btnScrollRight.Disable();
+            btnScrollLeft.Disable();
+        }
+        else
+        {
+            thumbWidth = Math.Max(width - (int)(width * nonDisplayedColumns / (double)Length),
+                MIN_BUTTON_WIDTH);
+
+            scrollablePixels = width - thumbWidth;
+
+            btnScrollRight.Enable();
+            btnScrollLeft.Enable();
+        }
+
+        buttonMinX = btnScrollLeft.Right + thumbWidth / 2;
+        buttonMaxX = Width - btnScrollRight.Width - (thumbWidth / 2);
+
+        RefreshButtonX();
+    }
+
+    /// <summary>
+    /// Scrolls the scrollbar when it's clicked on.
+    /// </summary>
+    public override void OnLeftClick()
+    {
+        base.OnLeftClick();
+
+        Scroll();
+    }
+
+    /// <summary>
+    /// Scrolls the scrollbar if the user presses the mouse left button
+    /// while moving the cursor over the scrollbar.
+    /// </summary>
+    public override void OnMouseMove()
+    {
+        base.OnMouseMove();
+
+        if (Cursor.LeftDown)
+        {
+            Scroll();
+            isHeldDown = true;
+            WindowManager.SelectedControl = this;
+        }
+    }
+
+    private void Scroll()
+    {
+        var point = GetCursorPoint();
+
+        if (point.X < btnScrollLeft.Width
+            || point.X > btnScrollRight.X)
```

Comment:
if (point.X < btnScrollLeft.**Width**
            || point.X > btnScrollRight.**X**)

is that correct or not?